dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -307,14 +308,16 @@ public void setLicenseInformation(Entry entry, String gitRepository) {
     }
 
     @Override
-    public boolean checkSourceCodeValidity() {
+    public boolean checkSourceCodeValidity(final boolean preload) {
         try {
             GHRateLimit ghRateLimit = github.getRateLimit();
             if (ghRateLimit.getRemaining() == 0) {
                 ZonedDateTime zonedDateTime = Instant.ofEpochSecond(ghRateLimit.getResetDate().getTime()).atZone(ZoneId.systemDefault());
-                throw new CustomWebApplicationException("Out of GitHub rate limit, please wait till " + zonedDateTime, HttpStatus.SC_BAD_REQUEST);
+                throw new CustomWebApplicationException(OUT_OF_GIT_HUB_RATE_LIMIT + zonedDateTime, HttpStatus.SC_BAD_REQUEST);
+            }
+            if (preload) {
+                github.getMyOrganizations();
```

Comment:
@denis-yuen Should we be doing this here? I kept it to not break any existing behavior. I guess you need some API call to verify the token is valid...

That particular call doesn't work for a GitHub app. The other option is to pick some API that works for both our Oauth and JWT tokens, if there is one, and get rid of the preload parameter I added.

Reply:
Looked, and I don't think there is another API call we can use that's available when auth is JWT.